pkg/zk/generators_test.go

```diff
@@ -222,6 +223,89 @@ var _ = Describe("Generators Spec", func() {
 					"example-annotation",
 					"example-value"))
 			})
+
+		})
+
+		Context("with pod policy securitycontext", func() {
+
+			BeforeEach(func() {
+				z := &v1beta1.ZookeeperCluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "example",
+						Namespace: "default",
+					},
+				}
+				z.WithDefaults()
+				z.Spec.VolumePermissions = true
+				no := int64(0)
+				securitycontext := v1.PodSecurityContext{
+					RunAsUser: &no,
+				}
+				z.Spec.Pod.SecurityContext = &securitycontext
+				z.Spec.InitContainers = []v1.Container{
+					{Name: "testcontainer2",
+						Image: "testimg1",
+					},
+				}
+				sts = zk.MakeStatefulSet(z)
+			})
+			It("should have runAsUser value as 0", func() {
+				Ω(fmt.Sprintf("%v", *sts.Spec.Template.Spec.SecurityContext.RunAsUser)).To(Equal("0"))
+			})
+		})
+
+		Context("with pod policy securitycontext runasuser empty", func() {
+
+			BeforeEach(func() {
+				z := &v1beta1.ZookeeperCluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "example",
+						Namespace: "default",
+					},
+				}
+				z.WithDefaults()
+				z.Spec.VolumePermissions = true
+				runAsNonRoot := false
```

Comment:
runAsRoot has been removed so either the test needs to be updated or removed if it's no long relevant.

Reply:
this unit test is added to cover the case runasuser is nil, but security context is not nil